controllers/auth/signup.js

```diff
@@ -0,0 +1,57 @@
+const jwt = require('jsonwebtoken');
+const randtoken = require('rand-token');
+
+const { Users, Tokens } = require('../../db/models');
+
+const key = process.env.secretKey;
+
+async function signup(req, res) {
+  const {
+    body: {
+      firstName,
+      lastName,
+      avatar,
+      email,
+      password,
+    },
+  } = req;
+  const regexp = /^([A-Za-z0-9_\-\.])+\@([A-Za-z0-9_\-\.])+\.([A-Za-z]{2,4})$/;
+  if (email.trim() === ''
+  || password === ''
+  || firstName.trim() === ''
+  || lastName.trim() === '') {
+    return res.status(400).send({ msg: 'Missed data.' });
+  }
+  const oldUser = await Users.findOne({ where: { email: email.trim() } });
+  if (oldUser) {
+    return res.status(409).send('user already exists');
+  }
+  if (!regexp.test(email)) {
+    return res.status(400).send({ msg: 'Invalid email.' });
+  }
+  try {
+    const user = await Users.create({
+      firstName: firstName.trim(),
```

Comment:
better to create payload with all applicable(trimmed if need) values and use it futher -> in condition and for user creation